src/utils/src/Semaphore.c

```diff
@@ -88,7 +113,12 @@ STATUS semaphoreCreateInternal(UINT32 maxPermits, PSemaphore* ppSemaphore)
     CHK(maxPermits > 0, STATUS_INVALID_ARG);
     CHK(NULL != (pSemaphore = (PSemaphore) MEMCALLOC(1, SIZEOF(Semaphore))), STATUS_NOT_ENOUGH_MEMORY);
     pSemaphore->maxPermitCount = maxPermits;
-    ATOMIC_STORE(&pSemaphore->permitCount, maxPermits);
+    pSemaphore->signalSemaphore = empty;
```

Comment:
For the same reasoning as above I chose against it. I agree that parameter changes could work fine, but this option allows us to avoid introducing a backwards compatibility issue for a commonly used library across many repos.

Reply:
My recommendation is to first think about your update story. Utility library was not versioned as it was going to be statically linked to higher-level libraries. It was later exposed publicly so it might be indeed be used by client applications so it needs to be backward compatible. In this case, an introduction of another API would be advisable - something like an ...Ex version which takes extra parameters in a form of a versioned struct (very heavy) or a better yet adding a version to the API - blahBlahV2 with it's own param set. All are pretty ugly but necessary for any public API